flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/hint/FlinkHints.java

```diff
@@ -189,18 +191,33 @@ public static RelNode capitalizeJoinHints(RelNode root) {
 
     private static class CapitalizeJoinHintShuttle extends RelShuttleImpl {
 
+        @Override
+        public RelNode visit(LogicalCorrelate correlate) {
+            return visitBiRel(correlate);
+        }
+
         @Override
         public RelNode visit(LogicalJoin join) {
-            List<RelHint> hints = join.getHints();
+            return visitBiRel(join);
+        }
+
+        private RelNode visitBiRel(BiRel biRel) {
+            Hintable hBiRel = (Hintable) biRel;
             AtomicBoolean changed = new AtomicBoolean(false);
             List<RelHint> hintsWithCapitalJoinHints =
-                    hints.stream()
+                    hBiRel.getHints().stream()
                             .map(
                                     hint -> {
                                         String capitalHintName =
                                                 hint.hintName.toUpperCase(Locale.ROOT);
                                         if (JoinStrategy.isJoinStrategy(capitalHintName)) {
                                             changed.set(true);
+                                            if (JoinStrategy.isLookupHint(hint.hintName)) {
```

Comment:
can you add some tests to verify this change

Reply:
please refer to `ClearJoinHintWithCapitalizeJoinHintShuttleTest`.